src/Compiler/Optimize/InnerLambdasToTopLevelFuncs.fs

```diff
@@ -219,8 +219,8 @@ module Pass1_DetermineTLRAndArities =
        let fArities = List.filter (fun (v, _) -> not (Zset.contains v rejectS)) fArities
        (*-*)
        let tlrS = Zset.ofList valOrder (List.map fst fArities)
-       let topValS = xinfo.TopLevelBindings                     (* genuinely top level *)
-       let topValS = Zset.filter (IsMandatoryNonTopLevel g >> not) topValS     (* restrict *)
+       let valReprS = xinfo.TopLevelBindings                     (* genuinely top level *)
+       let valReprS = Zset.filter (IsMandatoryNonTopLevel g >> not) valReprS     (* restrict *)
```

Comment:
Reverted this one - would need to review the terminology throughout this particular optimization phase - note the comment refers to "top" - in any case, we can just let this one be as it is for now